code/mapping_visualization/src/visualization.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python
+
+
+import ros_compatibility as roscomp
+from ros_compatibility.node import CompatibleNode
+from std_srvs.srv import SetBool, SetBoolRequest, SetBoolResponse
+from visualization_msgs.msg import Marker, MarkerArray
+from mapping.msg import Map as MapMsg
+
+from rospy import Publisher
+from rospy import Duration
+
+from mapping_common.entity import Entity
+from mapping_common.map import Map
+
+
+class Visualization(CompatibleNode):
+    """The visualization for the intermediate layer.
+
+    This Node will publish the marker array composed of the different entities.
+    """
+
+    def __init__(self, args):
+        super().__init__("mapping_visualization")
+
+        self.marker_publisher: Publisher = self.new_publisher(
+            MarkerArray, "/paf/hero/mapping/marker_array", qos_profile=1
+        )
+
+        self.filter_is_collider = False
+        self.filter_is_tracked = False
+        self.filter_is_stopmark = False
+        self.filter_is_lanemark = False
+        self.filter_is_ignored = False
+
+        self.new_service(SetBool, "vis/set_is_collider", self.set_is_collider_filter)
+        self.new_service(SetBool, "vis/set_is_tracked", self.set_is_tracked_filter)
+        self.new_service(SetBool, "vis/set_is_stopmark", self.set_is_stopmark_filter)
+        self.new_service(SetBool, "vis/set_is_lanemark", self.set_is_lanemark_filter)
+        self.new_service(SetBool, "vis/set_is_ignored", self.set_is_ignored_filter)
+
+        self.new_subscription(
+            topic=self.get_param("~map_topic", "/paf/hero/mapping/init_data"),
+            msg_type=MapMsg,
+            callback=self.map_callback,
+            qos_profile=1,
+        )
+
+    def map_callback(self, data: MapMsg):
+        map = Map.from_ros_msg(data)
+        marker_array = MarkerArray()
+
+        for id, entity in enumerate(map.entities):
```

Comment:
should we use map.entities_without_hero() here? people will read this as reference. then they immediately learn from this